src/05-derivatives/03-imaging.md

```diff
@@ -0,0 +1,367 @@
+  }
+}
+```
+
+This would be equivalent to having two JSON metadata files, one
+corresponding to `res-lo`
+(`pipeline1/sub-001/func/sub-001_task-rest_run-1_space-MNI305_res-lo_bold.json`):
+
+```JSON
+{
+  "SkullStripped": true,
+  "Resolution": "Matched with original BOLD resolution (2x2x3 mm^3)"
+}
+```
+
+And one corresponding to `res-hi`
+(`pipeline1/sub-001/func/sub-001_task-rest_run-1_space-MNI305_res-hi_bold.json`):
+
+```JSON
+{
+  "SkullStripped": true,
+  "Resolution": "Matched with high-resolution T1w (0.7mm, isotropic)"
+}
+```
+
+Example of CIFTI-2 files (a format that combines regularly sampled data
+and non-parametric surfaces) having both `res` and `den` keywords:
+
+```Text
+pipeline1/
+    sub-001/
+        func/
+            sub-001_task-rest_run-1_space-fsLR_res-1_den-10k_bold.dtseries.nii
+            sub-001_task-rest_run-1_space-fsLR_res-1_den-41k_bold.dtseries.nii
+            sub-001_task-rest_run-1_space-fsLR_res-2_den-10k_bold.dtseries.nii
+            sub-001_task-rest_run-1_space-fsLR_res-2_den-41k_bold.dtseries.nii
+            sub-001_task-rest_run-1_space-fsLR_bold.json
+```
+
+And the corresponding `sub-001_task-rest_run-1_space-fsLR_bold.json` file:
+
+```JSON
+{
+    "SkullStripped": true,
+    "Resolution": {
+        "1": "Matched with MNI152NLin6Asym 1.6mm isotropic",
+        "2": "Matched with MNI152NLin6Asym 2.0mm isotropic"
+    },
+    "Density": {
+        "10k": "10242 vertices per hemisphere (5th order icosahedron)",
+        "41k": "40962 vertices per hemisphere (6th order icosahedron)"
+    }
+}
+```
+
+## Masks
+
+Template:
+
+```Text
+<pipeline_name>/
+    sub-<participant_label>/
+        anat|func|dwi/
+            <source_keywords>[_space-<space>][_res-<label>][_den-<label>][_desc-<label>]_mask.nii.gz
+```
+
+A binary (1 - inside, 0 - outside) mask in the space defined by `<space>`. By
+default (i.e., if no transformation has taken place) the value of `space` should
+be set to `orig`.
+
+JSON metadata fields:
+
+| **Key name** | **Description**                                                                                                                                |
+| ------------ | ---------------------------------------------------------------------------------------------------------------------------------------------- |
+| RawSources   | Same as defined in [Introduction][intro], but elevated from OPTIONAL to REQUIRED                                                               |
+| Type         | RECOMMENDED. Short identifier of the mask. Reserved values: `Brain` - brain mask, `Lesion` - lesion mask, `Face` - face mask, `ROI` - ROI mask |
+| Resolution   | REQUIRED if `res` is present. String or dictionary of label to string. Specifies the interpretation of the resolution keyword.                 |
+| Density      | REQUIRED if `den` is present. String or dictionary of label to string. Specifies the interpretation of the density keyword.                    |
```

Comment:
I added `res`/`den` to masks and segmentations, as I think that's necessary.

And I think now I see where @edickie was going with a general top-level JSON file that didn't require a suffix, since this metadata is going to be the same for `bold`, `mask` and `dseg` files. While duplication isn't great, it's probably better than adding an entirely new filename convention.

At one point we discussed a `spaces.json`, which could make a useful short extension PR, but I think that should be backwards compatible, so doesn't need to be addressed before releasing derivatives.

Reply:
Please use the formal JSON terminology of `object` instead of `dict`, which is Pythonese. See https://www.json.org/json-en.html and https://github.com/bids-standard/bids-specification/pull/265#issuecomment-634276265